server/integration/testsuite/src/test/java/org/infinispan/server/test/jcache/JCacheStateAwarenessIT.java

```diff
@@ -0,0 +1,184 @@
+   private static final String enableManagementTestCacheName = "enableManagementTestCache";
+   private static final String getCacheNamesTestCacheName = "getCacheNamesTestCache";
+   private static final String getCacheTestCacheName = "getCacheTestCache";
+
+   private static final String testKey = "testKey";
+   private static final String testValue = "testValue";
+
+   final int managementPort = 9999;
+
+   @InfinispanResource("cachecontainer")
+   private RemoteInfinispanServer server1;
+   private RemoteCacheManager rcm1;
+
+   private CachingProvider jcacheProvider;
+   private CacheManager cacheManager;
+   private MBeanServer mBeanServer;
+
+   @Before
+   public void setUp() {
+      if (rcm1 == null) {
+         Configuration conf = new ConfigurationBuilder().addServer()
+               .host(server1.getHotrodEndpoint().getInetAddress().getHostName())
+               .port(server1.getHotrodEndpoint().getPort()).build();
+
+         rcm1 = new RemoteCacheManager(conf);
+
+         jcacheProvider = Caching.getCachingProvider("org.infinispan.jcache.remote.JCachingProvider");
+         cacheManager = jcacheProvider.getCacheManager();
+
+         mBeanServer = ManagementFactory.getPlatformMBeanServer();
+
+      }
+   }
+
+   /**
+    * Test enableStatistics command on JCache API <br>
+    * <br>
+    * Fix pending in <b>ISPN-7095</b>, needs to be uncommented when implemented
+    * 
+    */
+   // @Test
+   public void testEnableStatistics() throws Exception {
+      String mBeanName = getMBeanName(enableStatisticsTestCacheName);
+
+      cacheManager.enableStatistics(enableStatisticsTestCacheName, false);
+
+      assertTrue("Statistics register as enabled after disabling",
+            !(Boolean) mBeanServer.getAttribute(new ObjectName(mBeanName), "StatisticsEnabled"));
+
+      cacheManager.enableStatistics(enableStatisticsTestCacheName, true);
+
+      assertTrue("Statistics register as disabled after enabling",
+            (Boolean) mBeanServer.getAttribute(new ObjectName(mBeanName), "StatisticsEnabled"));
+
+   }
+
+   /**
+    * Test enableManagement command on JCache API<br>
+    * <br>
+    * Fix pending in <b>ISPN-7095</b>, needs to be uncommented when implemented
+    * 
+    */
+   // @Test
+   public void testEnableManagement() {
+      getMBeanName(enableManagementTestCacheName);
+   }
+
+   /**
+    * Enables management and retrieves canonical name of managed bean
+    * 
+    * @param testCacheName
+    *           name of the cache to enable management on
+    * @return canonical name of the managed bean
+    */
+   private String getMBeanName(String testCacheName) {
+
+      cacheManager.enableManagement(testCacheName, true);
+
+      Set<ObjectName> mBeanSet = mBeanServer.queryNames(null, null).stream()
+            .filter(n -> n.getCanonicalName().toUpperCase().contains(testCacheName.toUpperCase()))
+            .collect(Collectors.toSet());
```

Comment:
AFAICT after you call `enableManagement`, the cache will be registered both under `TestingUtil.getCacheObjectName()` and under `RIMBeanServerRegistrationUtility.calculateObjectName()`.